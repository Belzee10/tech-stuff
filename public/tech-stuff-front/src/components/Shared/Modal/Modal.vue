<template>
  <v-dialog v-model="isOpen" persistent width="500">
    <v-card>
      <v-card-title class="headline title" primary-title>
        {{ title }}
      </v-card-title>
      <v-divider></v-divider>
      <v-card-text class="pa-4">
        <slot></slot>
      </v-card-text>
      <v-card-actions class="justify-end">
        <v-btn
          :color="confirmButton.color"
          small
          class="text-capitalize mr-1 confirm"
          @click="() => $emit('confirm')"
        >
          {{ confirmButton.label }}
        </v-btn>
        <v-btn
          :color="cancelButton.color"
          small
          class="text-capitalize cancel"
          @click="() => $emit('cancel')"
          >{{ cancelButton.label }}</v-btn
        >
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script>
export default {
  name: 'Modal',
  props: {
    isOpen: {
      type: Boolean,
      default: false
    },
    title: {
      type: String,
      default: ''
    },
    confirmButton: {
      type: Object,
      default: () => ({
        color: '',
        label: ''
      })
    },
    cancelButton: {
      type: Object,
      default: () => ({
        color: '',
        label: ''
      })
    }
  }
};
</script>

<style lang="scss" scoped></style>
