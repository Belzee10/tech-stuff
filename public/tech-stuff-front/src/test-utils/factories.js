import { mount, createLocalVue } from '@vue/test-utils';
import mergeWith from 'lodash.mergewith';
import Vuetify from 'vuetify';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);

let vuetify;

beforeEach(() => {
  vuetify = new Vuetify();
});

const customizer = (_, srcValue) => {
  if (Array.isArray(srcValue)) {
    return srcValue;
  }
  if (srcValue instanceof Object && Object.keys(srcValue).length === 0) {
    return srcValue;
  }
};

/**
 * create component wrapper
 * @param {*} Component
 * @param {Object} overrides
 */
const createWrapper = (Component, overrides) => {
  const defaultMountingOptions = {
    localVue,
    vuetify
  };
  return mount(
    Component,
    mergeWith(defaultMountingOptions, overrides, customizer)
  );
};

/**
 * create Vuex store
 * @param {Object} overrides
 */
const createStore = overrides => {
  const defaultStoreConfig = {};
  return new Vuex.Store(mergeWith(defaultStoreConfig, overrides, customizer));
};

export { createWrapper, createStore };
