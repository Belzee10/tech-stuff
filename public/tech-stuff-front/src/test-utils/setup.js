import Vue from 'vue';
import Vuetify from 'vuetify';

Vue.config.productionTip = false;

Vue.use(Vuetify);

/**
 * silent "console.error"
 */
global.console = {
  error: jest.fn(),
  log: console.log,
  warn: console.warn,
  info: console.info,
  debug: console.debug
};
