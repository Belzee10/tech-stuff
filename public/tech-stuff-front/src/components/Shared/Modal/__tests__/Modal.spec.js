import { createWrapper } from '@/test-utils/factories';
import Modal from '../Modal.vue';

describe('Modal.vue', () => {
  test('should render correctly', () => {
    const wrapper = createWrapper(Modal);
    expect(wrapper.element).toMatchSnapshot();
  });
});
