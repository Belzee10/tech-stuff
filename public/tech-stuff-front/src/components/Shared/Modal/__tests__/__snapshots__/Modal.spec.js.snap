// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal.vue should render correctly 1`] = `
<div
  class="v-dialog__container"
  role="dialog"
>
  <div
    class="v-dialog__content"
    role="document"
    style="z-index: 0;"
  >
    <div
      class="v-dialog v-dialog--persistent"
      style="width: 500px; display: none;"
    />
  </div>
</div>
`;
